File: android/app/src/main/java/com/mochi/keyboard/data/AppContainer.kt
package com.mochi.keyboard.data

import android.content.Context
import com.google.firebase.auth.FirebaseAuth
import com.google.firebase.firestore.FirebaseFirestore
import com.google.firebase.functions.FirebaseFunctions
import com.google.firebase.storage.FirebaseStorage

/** Manual DI container — no framework (Hilt/Koin) yet; small enough app that it isn't worth the
 * setup ceremony today. Revisit if the repository count grows enough to make this wiring painful. */
class AppContainer(
    context: Context,
    auth: FirebaseAuth,
    firestore: FirebaseFirestore,
    functions: FirebaseFunctions,
    storage: FirebaseStorage = FirebaseStorage.getInstance()
) {
    val userRepository: UserRepository = UserRepository(firestore)
    val authRepository: AuthRepository = AuthRepository(auth, functions, userRepository)
    val themeRepository: ThemeRepository = ThemeRepository(firestore)
    val wallpaperRepository: WallpaperRepository = WallpaperRepository(firestore)
    val wallpaperLibraryRepository: WallpaperLibraryRepository = WallpaperLibraryRepository(context)
    val likeRepository: LikeRepository = LikeRepository(firestore)
    val followRepository: FollowRepository = FollowRepository(firestore)
    val blockRepository: BlockRepository = BlockRepository(firestore)
    val reportRepository: ReportRepository = ReportRepository(firestore)
    val createRepository: CreateRepository = CreateRepository(firestore)
    val storageRepository: StorageRepository = StorageRepository(storage)
    val settingsRepository: SettingsRepository = SettingsRepository(context)
    val searchHistoryRepository: SearchHistoryRepository = SearchHistoryRepository(context)
    val billingRepository: BillingRepository = BillingRepository(context).apply {
        configure(appUserId = auth.currentUser?.uid)
    }
}

File: ios/MochiApp/Data/AppContainer.swift
import FirebaseAuth
import FirebaseFirestore
import FirebaseFunctions
import FirebaseStorage

/// Manual DI container, same "no framework yet" call android/.../data/AppContainer.kt made —
/// `shared` is nil whenever `FirebaseEnvironment.configureIfPossible()` couldn't run (no
/// GoogleService-Info.plist in the bundle yet), so every screen that wants real data must check for
/// nil and fall back to its existing MockData presentation rather than force-unwrapping.
final class AppContainer {
    static let shared: AppContainer? = {
        guard FirebaseEnvironment.configureIfPossible() else { return nil }
        return AppContainer()
    }()

    let auth: Auth
    let firestore: Firestore
    let functions: Functions
    let storage: Storage

    let userRepository: UserRepository
    let authRepository: AuthRepository
    let themeRepository: ThemeRepository
    let likeRepository: LikeRepository
    let followRepository: FollowRepository
    let reportRepository: ReportRepository
    let createRepository: CreateRepository
    let storageRepository: StorageRepository

    private init() {
        auth = Auth.auth()
        firestore = Firestore.firestore()
        functions = Functions.functions()
        storage = Storage.storage()

        userRepository = UserRepository(firestore: firestore)
        authRepository = AuthRepository(auth: auth, functions: functions, userRepository: userRepository)
        themeRepository = ThemeRepository(firestore: firestore)
        likeRepository = LikeRepository(firestore: firestore)
        followRepository = FollowRepository(firestore: firestore)
        reportRepository = ReportRepository(firestore: firestore)
        createRepository = CreateRepository(firestore: firestore)
        storageRepository = StorageRepository(storage: storage)
    }
}
